Add alt text to JupyterLab documentation

isabela-pf opened the pull request:
# Welcome to the working PR!

This is the working PR for the SciPy sprint alt text workshops! This is where we collaborate during the event before we send a clean contribution over to the [main JupyterLab repo](https://github.com/jupyterlab/jupyterlab).

For the full info about this event, please visit the [event agenda] (to be linked).

## How to contribute

⭐ For full info about how to interact with this PR, please visit the [How to Participate section of the agenda] (to be linked). Please contribute via [suggestions](https://hackmd.io/@marslee/BksnYfWVK) so we can easily review each other's work and give you credit in the commit history!

✏️ If you plan to work on a file, let people know by commenting below. You can do this as a group, and you can write alt text for as many images as you'd like. We can have multiple suggestions for the same alt text, but it would be nice if we spread out to cover more images.

📚 If you need a reference for how to format your suggestions, [alt text in Markdown files (ends with `.md`) follow this format](https://www.markdownguide.org/basic-syntax/#images-1) while [alt text in RST files (ends with `.rst`) follow this format](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/images.html).

🖼️ There are two types of images in this PR: ones that already have alt text and ones that have none. 
- Images that appear in this PR with existing alt text are still worth reviewing and editing! If you think they could benefit from a different description, please suggest one.
- Images with no alt text need a description added in place of the blank.

❗ If you think an image is missing from these files, let @isabela-pf know and they will add it to the PR.

## Images in this PR

All images with prompts to add alt text are rendered as one of the following links. They are listed in the same order as you will find them in the Files Changed tab on this PR.

1. JupyterLab README: `README.md` ([Link to rendered readme](https://github.com/jupyterlab/jupyterlab/#readme))
2. Contributing to JupyterLab:`docs/source/developer/contributing.rst` ([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/developer/contributing.html))
3. Internationalization and Localization: `docs/source/developer/internationalization.rst` ([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/developer/internationalization.html))
4. Notebook: `docs/source/extension/notebook.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/extension/notebook.html))
5. Overview: `docs/source/getting_started/overview.rst` ([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/getting_started/overview.html))
6. JupyterLab Documentation: `docs/source/index.rst` ([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/index.html))
7. Command Palette:`docs/source/user/commands.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/commands.html))
8. Debugger:`docs/source/user/debugger.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/debugger.html))
9. Exporting Notebooks:`docs/source/user/export.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/export.html))
10. Extensions:`docs/source/user/extensions.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/extensions.html))
11. Text Editor:`docs/source/user/file_editor.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/file_editor.html))
12. File and Output Formats:`docs/source/user/file_formats.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/file_formats.html))
13. Working with Files:`docs/source/user/files.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/files.html))
14. The JupyterLab Interface:`docs/source/user/interface.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/interface.html))
15. Localization and Language:`docs/source/user/language.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/language.html))
16. Notebooks:`docs/source/user/notebook.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/notebook.html))
17. Real Time Collaboration:`docs/source/user/rtc.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/rtc.html))
18. Managing Kernels and Terminals:`docs/source/user/running.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/running.html))
19. Terminals:`docs/source/user/terminal.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/terminal.html))
20. Table of Contents:`docs/source/user/toc.rst`([Link to rendered docs](https://jupyterlab.readthedocs.io/en/stable/user/toc.html))
21. json-extension package README: `packages/json-extension/README.md` ([Link to rendered packaged readme](https://github.com/jupyterlab/jupyterlab/tree/master/packages/json-extension#readme)).
22. JupyterLab Services package README: `packages/services/README.md` ([Link to rendered readme](https://github.com/jupyterlab/jupyterlab/tree/master/packages/services#readme)).
23. JupyterLab Status Bar package README: `packages/statusbar-extension/README.md` ([Link to rendered packaged readme](https://github.com/jupyterlab/jupyterlab/tree/master/packages/statusbar-extension#readme)).
24.vdom-extension package README: `packages/vdom-extension/README.md` ([Link to rendered packaged readme](https://github.com/jupyterlab/jupyterlab/tree/master/packages/vdom-extension#readme)).
25. vega5-extension package README: `packages/vega5-extension/README.md` ([Link to rendered packaged readme](https://github.com/jupyterlab/jupyterlab/tree/master/packages/vega5-extension#readme)).

<!-- 
- [ ] 1. Line number (or other identifying feature)
- [ ] 2. Line number (or other identifying feature)
- [ ] 3. Line number (or other identifying feature)
-->
---

If you need anything, please ask @isabela-pf 🌻

isabela-pf committed:
Add white space to readme images

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#12741)

updates:
- [github.com/sirosen/check-jsonschema: 0.16.0 → 0.16.2](https://github.com/sirosen/check-jsonschema/compare/0.16.0...0.16.2)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

jtpio committed:
Update dev dependencies (#12698)

* Update dependencies

* Stay on `eslint-plugin-react@~7.29.4`

* Fix stylelint rules

* Try with import-notation: null

vidartf committed:
Fix staging/yarn.lock registry (#12742)

* Fix  UnhandledPromiseRejectionWarning

Future proofing for node.

* Fix broken replace on Windows

'\\/g' matches very little :)

* Add integrity check for lockfile registries

We rely on the registry of resolved lockfile entries in commands.py, so we should ensure that the entries in the staging lockfile all use the default one. Otherwise we should change the logic in commands.py to handle multiple registries.

* Fix regex

* Fix registry in root yarn.lock manually

vidartf committed:
Optimize `eachLine` loops (#12746)

When modifying each line of a large document it is very important that we do not update the DOM after each update. In CodeMirror this is avoided by wrapping things in operations.

For `clearGutter` we just call the already defined method in CM, which uses an operation by default. It might also be slightly more optimized in other ways.

martinRenou committed:
Debugger: Fix CSS for variables inspecting (#12749)

krassowski committed:
Bump marked version (#12747)

Co-authored-by: Frédéric Collonval <fcollonval@users.noreply.github.com>

dependabot[bot] committed:
Bump shell-quote from 1.7.2 to 1.7.3 (#12744)

Bumps [shell-quote](https://github.com/substack/node-shell-quote) from 1.7.2 to 1.7.3.
- [Release notes](https://github.com/substack/node-shell-quote/releases)
- [Changelog](https://github.com/substack/node-shell-quote/blob/master/CHANGELOG.md)
- [Commits](https://github.com/substack/node-shell-quote/compare/v1.7.2...1.7.3)

---
updated-dependencies:
- dependency-name: shell-quote
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vidartf committed:
Debugger: Make kernel source list react based (#12751)

* Make kernel sources filter sticky

This way it will be visible even if you scroll a lot!

* Make kernel source list react based

This improves performance when updating of the kernel sources, which happens whenever a document that is being debugged gets focused. This includes when switching from the notebook document to a kernel source editor. Ideally the kernel sources would not be refreshed when switching between tabs that share the same debug context.

peytondmurray committed:
Add resize handle for scrolled cell outputs (#12740)

yuvipanda committed:
Bump xtermjs to latest (#12715)

* Bump xtermjs to latest

Ref https://github.com/jupyterlab/jupyterlab/issues/10990

Seems to improve deadkeys behavior in Safari but not Firefox

* Update Playwright snapshots

Co-authored-by: Frédéric Collonval <fcollonval@gmail.com>

fcollonval committed:
Update snapshots for challenger commit

fcollonval committed:
Set focus when active cell changes only from mouse click (#12735)

* Set focus when active cell changes only from mouse click
Fixes #12723

* Update Playwright Snapshots

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

fcollonval committed:
Remove unstubExtensionsSearch (#12738)

isabela-pf committed:
Add alt attributes and white space

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#12771)

updates:
- [github.com/psf/black: 22.3.0 → 22.6.0](https://github.com/psf/black/compare/22.3.0...22.6.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump parse-url from 6.0.0 to 6.0.2 (#12773)

Bumps [parse-url](https://github.com/IonicaBizau/parse-url) from 6.0.0 to 6.0.2.
- [Release notes](https://github.com/IonicaBizau/parse-url/releases)
- [Commits](https://github.com/IonicaBizau/parse-url/commits)

---
updated-dependencies:
- dependency-name: parse-url
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fcollonval committed:
Fix memory leaks (#12750)

* Release most objects following a notebook creation

* Clear more signals

* Clear properly open with submenu

* Dispose correctly resources

* Fix unit tests

* Stop debugger on widget dispose

fcollonval committed:
remove reference to blueprint.js in css (#12663)

* remove reference to blueprint.js in css

* remove more blueprint.js overrides

* add back the color and font-size rules for the body

* Fix Card styles

* Restore unset outline on focus

* Update Playwright Snapshots

Update Playwright Snapshots

Update Playwright Snapshots

Update snapshots

* Update snapshots

Co-authored-by: Eric Charles <eric@datalayer.io>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

hbcarlos committed:
Updates ydoc (#12779)

* Updates ydoc

* Update setup.cfg

Co-authored-by: David Brochart <david.brochart@gmail.com>

Co-authored-by: David Brochart <david.brochart@gmail.com>

fcollonval committed:
Bump @lumino/widgets to 1.33.0 (#12777)

* Bump @lumino/widgets to 1.33.0

* Undo change in staging folder

fcollonval committed:
Removes lighthouse and markdown-loader-jest (#12776)

* Removes lighthouse and markdown-loader-jest

* Fix node type errors

* Update documentation

* Convert MD to RST

fcollonval committed:
[memory-leaks] Fixes following cell addition analysis (#12774)

* disconnect properly the cell toolbar

* Remove `StaticNotebook._cellsArray`

* Ensure to dispose debouncer

dependabot[bot] committed:
Bump moment from 2.29.2 to 2.29.4 (#12781)

Bumps [moment](https://github.com/moment/moment) from 2.29.2 to 2.29.4.
- [Release notes](https://github.com/moment/moment/releases)
- [Changelog](https://github.com/moment/moment/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/moment/moment/compare/2.29.2...2.29.4)

---
updated-dependencies:
- dependency-name: moment
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

siddartha-10 committed:
Update base.css (#12783)

Co-authored-by: siddartha1018 <72271363+siddartha1018@users.noreply.github.com>

eirrgang commented on docs/source/developer/contributing.rst:
```diff
@@ -758,6 +759,7 @@ The Debugger Adapter Protocol
 The following diagram illustrates the types of messages sent between the JupyterLab extension and the kernel.
 
 .. image:: ./debugger_protocol_diagram.png
+   :alt: 
```
```suggestion
   :alt: UML sequence diagram illustrating the interaction between a user, JupyterLab, and the kernel. From top to bottom, the timeline starts with opening the notebook and includes annotations where the debugger is started and stopped. Specific interactions and message types are discussed in the subsequent text.
```

eirrgang commented on docs/source/developer/contributing.rst:
```diff
@@ -767,6 +769,7 @@ Inspecting the debug messages in VS Code can be useful to understand when debug
 The first step is to create a test file and a debug configuration (``launch.json``):
 
 .. image:: ./debugger_launch_configuration.png
+   :alt: 
```
```suggestion
   :alt: Small screenshot of the GUI showing the menu for creating a debug configuration.
```

isabela-pf committed:
Add event-contributed alt text!

Co-authored-by: M. Eric Irrgang <mei2n@virginia.edu>

isabela-pf committed:
Add event-contributed alt text!

Co-authored-by: M. Eric Irrgang <mei2n@virginia.edu>

mwcraig commented on docs/source/developer/contributing.rst:
```diff
@@ -488,6 +488,7 @@ Now run Lighthouse against this local server and show the results:
    jlpm run lighthouse --view
 
 .. image:: ../images/lighthouse.png
+   :alt: 
```
```suggestion
   :alt: Screenshot of Lighthouse tool used for benchmarking performance.
```

mwcraig commented on docs/source/getting_started/overview.rst:
```diff
@@ -8,6 +8,7 @@ JupyterLab is a next-generation web-based user interface for Project Jupyter.
 .. image:: ../images/interface-jupyterlab.png
    :align: center
    :class: jp-screenshot
+   :alt: 
```
```suggestion
   :alt: Screenshot of Jupyter lab interface showing the file browser, and notebook and several other open files.
```

isabela-pf committed:
Add event-contributed alt text!

Co-authored-by: Matt Craig <mattwcraig@gmail.com>

isabela-pf committed:
Add event-contributed alt text!

Co-authored-by: Matt Craig <mattwcraig@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#12796)

updates:
- [github.com/sirosen/check-jsonschema: 0.16.2 → 0.17.0](https://github.com/sirosen/check-jsonschema/compare/0.16.2...0.17.0)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

fcollonval committed:
Use Vega SVG renderer to drop canvas dependency (#12785)

* Use SVG renderer in Vega to drop canvas dependency
Fixes #12726

* Fix tests

* Update Playwright Snapshots

* Rename variables

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

RobbyPratl committed:
update tab name after file rename (#12791)

Co-authored-by: Robert Pratl <robertpratl@roberts-mbp.attlocal.net>

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#12813)

updates:
- [github.com/sirosen/check-jsonschema: 0.17.0 → 0.17.1](https://github.com/sirosen/check-jsonschema/compare/0.17.0...0.17.1)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

JohanMabille committed:
Migrate to Codemirror 6 (#11638)

* Upgraded to codemirror 6 in package.json files

Fix find package.json

Fix find package.json

* Removed codemirror extra style

* Upgraded to CodeMirror6: first pass

* Fixed the build

* Added back yjs bindings

* Added modes for syntax highlighting back

* Added dynamically loaded dependencies to the list of unused dependencies

* Plugged new modes into editor

* Workaround for missing declaration of mkPython in legacy-modes

Rebased on master

Generated yarn.lock

Package integrity updates

Generated yarn.lock

* Implemented centralized configuration for the editor

* Rerouting language and key bindings changes to EditorConfiguration

* Added style back

* Migration of debugger package to CodeMirror 6

* Removed codemirrorsearchprovider.ts

Rebased

Regenerated yarn.lock

* Migrated search feature to CM6

* code cleanup

* Added listener to fileeditor search provider

* Fix Jest setup

* Fix React type

* Fix benchmark tests

* Upgrade @codemirror 0.19.x

* Fix integrity

* Fix prettier

* Fixed fileeditor search

* Migrated selection marks to codemirror 6

* Code cleanup

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Added event handler for keydown event

* Added markdown highlighting

* linter

* Fixed editor tests

* Fixed codeeditor test suite

* Fixed docregistry test suite

* Fixed debugger test suite

* Upgraded to CodeMirror 6 (was beta before)

* Fixed auto indent and tab indent

* Fixed fileeditor css

* linter and yarn.lock

* Update yarn.lock

* Tracked last change in editor and code clean up

* Fixed gutter background in dark theme

* Ported theme and syntax highlighting to CM6

* Added map for themes

* Regenerated yarn.lock

* Applied changes from review

* Generated yarn.lock after rebase

* Migrated getToken feature to CM6

* Fixed examples

* Fixed galata tests

* Update Playwright Snapshots

* Update Playwright Snapshots

* Triggers CI

* Applied review

* Update snapshots

* Additional clean-up following removal of `editor.refresh`

* Apply code block parsing in md

* Fix integrity

* Clean up theme menu

* Display trailing unstyled code in markdown viewer

* Fixed codeeditor test

* Fix handling of brainf... language

* Fix highlithing in rendered markdown code blocks

* Update Playwright Snapshots

* Update snapshot

Co-authored-by: Frédéric Collonval <fcollonval@gmail.com>
Co-authored-by: Jeremy Tuloup <jeremy.tuloup@gmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

isabela-pf committed:
Add event-contributed alt text!

Co-authored-by: Aerik Pawson <45904740+aerikpawson@users.noreply.github.com>

jtpio committed:
Update `verdaccio`, start registry on `0.0.0.0` instead of `localhost` (#12799)

* Install `playwright@~1.21.0` in the browser check

* Install @playwright/test

* Install playwright

* Handle logged on stdout first

* Try with yarnpkg

* debug

* debug

* try installing another package

* Install the latest playwright

* Try with npm

* Set the registry through env var

* Undo yarnpkg changes

* Increase timeouts

* Try with network-timeout

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* jlpm config

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add uplink timeout

* update verdaccio config

* quotes

* Bump to the latest `verdaccio`

* Remove timeout field

* indent

* fix typo

* pass registry explicitely

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* try with 0.0.0.0

* Revert explicit registry

* Update yarn.lock

Co-authored-by: Frédéric Collonval <fcollonval@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Frédéric Collonval <fcollonval@gmail.com>

fcollonval committed:
Galata `tmpPath` does not work on Windows
Fixes #12822

dependabot[bot] committed:
Bump terser from 4.8.0 to 4.8.1 (#12818)

Bumps [terser](https://github.com/terser/terser) from 4.8.0 to 4.8.1.
- [Release notes](https://github.com/terser/terser/releases)
- [Changelog](https://github.com/terser/terser/blob/master/CHANGELOG.md)
- [Commits](https://github.com/terser/terser/commits)

---
updated-dependencies:
- dependency-name: terser
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

afshin committed:
Merge pull request #12823 from fcollonval/fix/12822-Galata-`tmpPath`-does-not-work-on-Windows

Use path to extract `tmpPath`

afshin committed:
Merge pull request #12820 from jupyterlab/12758-snapshots-are-not-updated-on-the-challenger-commit-for-benchmark-tests

Update snapshots for challenger commit

fcollonval committed:
Robuster UI tests (#12821)

* Robuster UI tests

* Fix reporter tests

* Improve robustness

* Update Playwright Snapshots

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

hbcarlos committed:
Remove drive prefix from the file path when creating the new path (#12824)

JasonWeill committed:
Adds version maintenance policy (#12829)

* Adds version maintenance policy

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* support -> maintain

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

fcollonval committed:
Automated Changelog Entry Forward Ported from v3.4.4 (#12837)

Co-authored-by: GitHub Action <41898282+github-actions[bot]@users.noreply.github.com>

thomasaarholt committed:
Adjust css to not leave trace of deleted widgets (#12838)

* Adjust css to not leave trace of deleted widgets

* Fix missing `.` in class selector

Co-authored-by: Frédéric Collonval <fcollonval@users.noreply.github.com>

yangql176 committed:
feat: Add .webp filetype in docRegistry. (#12839)

* fix: select kernal when open a .ipynb file is not translated correctly  (#12133)

* fix: Markdown cell generates duplicate toc content (#12312

* update fork

* feat: Add .webp filetype in docRegistry (#12832)

Co-authored-by: yangqinliu <yangqinliu@baidu.com>
Co-authored-by: yangqinliu <yangqinliu@Iris-MacBook-Pro.local>

JasonWeill committed:
Removes info about meeting notes on Binder (#12847)

davidbrochart committed:
Move YDocWebSocketHandler to jupyter_server_ydoc (#12772)

fcollonval committed:
Fix CI failures (#12843)

* Fix browser-require example

* Fix UI test and freeze network request for robustness

* Increase example timeout

* Anonymous dump

dependabot[bot] committed:
Bump toshimaru/auto-author-assign from 1.5.0 to 1.5.1 (#12854)

Bumps [toshimaru/auto-author-assign](https://github.com/toshimaru/auto-author-assign) from 1.5.0 to 1.5.1.
- [Release notes](https://github.com/toshimaru/auto-author-assign/releases)
- [Changelog](https://github.com/toshimaru/auto-author-assign/blob/main/CHANGELOG.md)
- [Commits](https://github.com/toshimaru/auto-author-assign/compare/v1.5.0...v1.5.1)

---
updated-dependencies:
- dependency-name: toshimaru/auto-author-assign
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump tj-actions/changed-files from 23 to 24 (#12853)

Bumps [tj-actions/changed-files](https://github.com/tj-actions/changed-files) from 23 to 24.
- [Release notes](https://github.com/tj-actions/changed-files/releases)
- [Changelog](https://github.com/tj-actions/changed-files/blob/main/HISTORY.md)
- [Commits](https://github.com/tj-actions/changed-files/compare/v23...v24)

---
updated-dependencies:
- dependency-name: tj-actions/changed-files
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

meeseeksmachine committed:
Backport PR #12855: Don't suggest deprecated command (#12857)

Co-authored-by: Ryan Lovett <rylo@berkeley.edu>

davidbrochart committed:
Store Y updates (#12852)

* Store Y updates

* Require jupyter_server_ydoc>=0.1.8,<0.2.0

jtpio committed:
Run `yarn-deduplicate` on `build:core` (#12850)

* Run `yarn-deduplicate` on `build:core`

* Update package.json

fcollonval committed:
Run memory-leak tests on PR (#12789)

* Run memory-leak tests on PR

* Fix server url

* Upload and publish memory leak report

* Fix saving memory leak report

* Remove trailing whitespace

* Use action output to save memory leak report

* Merge memory leak artifact with benchmark ones

* Clean the changes following output directly as artifact

hsuanxyz committed:
Fix kernel in the statusbar does not match the actual (#12865)

fix #12592

trungleduc committed:
Set `Contents.ContentType` to string

oscar6echo committed:
Add trustbydefault param to htmlviewer-extension (#12868)

* Add trustbydefault param to htmlviewer-extension

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Run prettier

* Use common pattern to check for settingRegistry before using it

* Rename trustByDefault parameter and variable

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

davidbrochart committed:
Bump jupyter_server_ydoc>=0.1.9 (#12876)

isabela-pf committed:
Post event clean up and copyediting

jtpio committed:
Update `yjs-codemirror.next`

Co-Authored-By: Johan Mabille <johan.mabille@gmail.com>
Co-Authored-By: Afshin T. Darian" <git@darian.email>

afshin committed:
Merge pull request #12880 from jtpio/y-codemirror

Update `yjs-codemirror.next`

SylvainCorlay committed:
Document building JupyterLab on osx-arm64 platforms (#12882)

fcollonval committed:
Merge branch 'master' into altscipy

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

SylvainCorlay committed:
Remove reference to unmaintained nb_conda_kernels (#12878)

hsuanxyz committed:
Fix progress bar not working after uploading multiple files finished (#12871)

fix #12798

trungleduc committed:
Integrate `jupyterlab-lsp` into `jupyterlab` (#12534)

* Import jumpers from index, update yarn

* Do not attach "jump back" command to context menu

* Flip the jump back icon in palette, remove debug log

* Implement Accel and use it as default to support ⌘  on Mac

* Add highlights debouncer and check if highlights request is needed

before sending one.

* Clear `last_token` after clearing highlights

* Ensure that trigger characters auto-invoke works in continuous hinting mode

* Add highlights test and fix #431

* Allow to clear highlights on blur

* Set promise type to void

* Fix syntax highlighting (needs import of cm/mode/meta)

* Try to fix R double colon completion issue

* Fix test and lint; add changelog

* Add support for jupyterlab-classic

* Make the status item work with classic on toolbar

* Fix highlight of status icon when selected

* Use new icons and no text on toolbar when status bar not available

* Disable diagnostics panel command in Classic

* Update yarn.lock, lint

* Fix connection manager loosing track of notebooks

which were not correctly scoped by path by switching away from
id_path which was refactored to not contain the actual file name
to uri which does contain the file name and path

* Add new status "no server extension"

* Fix handling of kernel change and restart

* Retry connection for 5.5 minutes total rather than 40 seconds

* Only claim that servers are missing if they really are

* Rewire all logs to go through a thin scoped wrapper

* Fix two typos

* Temporarily tolerate pygls usage of nulls for diagnostics

* Fix diagnostics panel not working after restart

* Ensure that diagnostics panel works after restart

* Remove prefixes

* Implement completionItem/resolve and icon substitution

* Move rendering to renderer

* Pre-fetch completion items (above/below), show placeholder while waiing

* Fix hiding of panel, fix illegal constructor for plain-text documentation

* Do not check item inset text if empty

* Fix Julia language server crashes on too broad save notification

* Make the LSP completions work when kernel is busy or slow

* Do not show delayed responses if line changed or cursor retreated

* Fix state not being updated after notebook rename/restart

* Implement allSettled replacement to show _something_ when one of

the completion sources fails

* Restore pre-formatted rendering for completer panel, enable latex

* Invalidate delayed completion if cell changed or lost focus

* Invalidate signature on cell blur, change and some cursor moves

* Fix comment

* WIP: enable sorting, improve marking and filtering and doc panel

* Remove outdated comment

* Allow tab-invoked completion in strings, suppress autoinvoke in def

* Fix dictionary completion issue for jedi-language-server

that would have led to "Code Editor out of Sync" error

* Prevent crash of the diagnostics panel when foreign documents are empty

* Fix diagnostics not being cleared upon foreign document closure

* Only search for label match if different from insertText match

and escape HTML label when no query is used

* Collapse paths for files deep in subfolders

* Fix source install (#526)

* Add a job to verify source install

* Add @jupyterlab-classic/application to peer dependencies

* Adding as peer-dependency did not help, entering debug mode

* Try to use peerDependenciesMeta

* Remove peerDependency(Meta) for classic

* Revert to safe use of app.shell with type casing and type import

* Restore smoke job

* Add changelog entry

* Re-activate atest

* Handle case where getCursor() returns null in highlighter

* Clean up null check

* Fix a typo preventing pinfo from working on ids with letter 's'

* Initial work on source-aware completion item and reply

* Add ipython-bigquery magic

* Add ipython-bigquery to defaults.ts

* update ipython-bigquery spec : comment %bigquery line magic

* make strings localizable

* Delete commented code

* Delete commented code

* update transclusions, run jlpm lint

* fix error in tests with rename.ts

* Fix regular-expression based extractors offset

* Lint

* Remove incorrect comment

* Swap order of transclusions provider so that generic IPython is last

* propose Extractor API

* Update the bigquery extractor to use foreign_capture_group

* Add key overrides exports, and a note

* foreign_capture_group: number →  foreign_capture_groups: number[]

* start adding example package with tests

* more work on example

* linting

* clobber some lint

* resolve

* Expose caseSensitive and includePerfectMatches completer settings

* Display 'source' or 'detail' next to completion label

* fix more lint, use built-in jest test data iteration

* reorganize API definitions to avoid circular imports

* Implement display of detail, source, and extra settings

* Do update extra info on render too

* fix code stype with lint

* fix test errors with diagnostics.ts

* localize Severity levels and columns menu of diagnostics panel

* fix code style

* use ICodeMirror, more type imports, order plugin for fixer

* Lint

* Defer translation of the status

* Allow to disable LSP or Kernel completions independently.

Some use love LSP for many of its features, but prefer to keep
completions via the kernel.

This adds two boolean flags in the configuration to not send completions
request to LSP and not send completions requests to the kernel.

See #440

* Move to exclusion list for config.

Take reviews into accounts

* Rename options

* prettier

* Add priority setting for servers handling the same language

* Fix failing test, update changelog, add serverIdentifier

* Address review comments, back down on lowercase

* Move the workaround to the root of the issue

* Special case for MacOS

* Fix this binding for priority detection

* Fix order of config initialization

* Normalize colons in Windows paths

* Implement console log window messages, update lsp-ws-connection

* Add server identifier

* JupyterLab Classic -> RetroLab

* Rework LSP connection interface, cherry picking #278

Co-authored-by: bollwyvl
Co-authored-by: krassowski

* Pass connection to server request handlers as well (as optional)

* Only emit priority warning once

* Fix registration options, diagnostics (one slot!), migrate hover

* Implement setTrace, logTrace, and optional logging of messages using new API

* Optimize lsp server error message (#621)

* Optimize lsp server error message

* Fix

* Update changelog

* Add types-six for new mypy version

* Add types-six to lint.yml

* Try installing mypy types in action

* Try to install types manually

* Might as well use requirements/lint.yml...

* Stop conda nagging about pip

* Lint lint.yml

Co-authored-by: zhouwenhua02 <zhouwenhua02@meituan.com>
Co-authored-by: krassowski <krassowski.michal@gmail.com>

* Fix hover malfunction messages (#630)

* Fix hover malfunction messages

* Sync completion-theme version, relax some constraints

* Update changelog, bump theme versions (as pin was relaxed)

* Add troubleshooting pop-up

* Fix tests, improve UI, add more troubleshooting

* Prepare for testing with Klingon

* Correct rpy2 `%Rdevice` magic integration (#646)

and prevent breaking of other line magics which also happen to start with R.

* Do not hide tooltip on mouseleave/move if close to tooltip

* Display MarkedStrings correctly as markdown in hover

Previously those were incorrectly shown as plain text strings

* Workaround breaking change in JupyterLab 3.1

* Fix typos, lint

* Handle case where position is null

* Add console attribute to VirtualDocument

* fix tests

* Use current version of CodeMirror typings

* Implement persistent signature, with active parameter highlight

* Improve styling of the signature

* Hoist closeCharacters to a setting, hide tooltip on blur, fix escapeMarkdown

* Implement hiding on cursor moving prior to start position

* Remove useless escape

* Escape backslash

* Implement remaining behaviour

* Lint

* Workaround incorrect initial placement of the tooltip

* Test signature to markdown, fix whitespace handling, use new API

* Fix backslash escaping

* Add browser tests for signature marks and closing

* Implement collapsing of long signatures

* Improve translations in diagnostics, update translation domain

* Ensure `trans` always available, camelCase some protected members

* Keep `get_breadcrumbs` as deprecated to avoid breaking compatibility

in case if anyone uses it

* Fix the issue with non-disappearing signature

* Fix kernel completions filtering for R

* Fix tab completions not showing up in strings due to incomplete trigger kind invalidation

* Fix path completion in strings

* Improve the completion test

* Soft rename of the organization

* Improve escaping for IPython magics overrides

* Allow to disable specific features

* Fix placeholder removal overwriting documentation from newer request

* Further improve documentation panel reliability

* Prettier

* Shim `workspace/configuration` to support new YAML server version

* issue 695
copy generated type files into lib directory

* Implement details-below layout

* Correct gettext method

* Fix translations for popup

* Fix `Line:Ch` and `Cell` sort order

* Test sorting of columns in diagnostic panel

* Fix identifier for pluralized strings

* Squash warnings and errors in console

* Enable strict null checks

* Enable `noImplicitThis`

* Fix hover character comparison logic, finish null position handling

* Only fetch position on actual user-commanded jump

* Handle an warning showing up on restoring workspace

* Enable `strictBindCallApply`

* Enable `noImplicitReturns`

* Fix: make signature (and details) clickable, correct signature

formatting when no arguments present

* Enable simple pre-filtering of completion matches

* Address test failures

* Add support for diagnostic tags from LSP 3.15

* Move `initializeParams` which are fully responsibility of client,

and not the connection implementation detail

* Announce tag support for diagnostics, correct capabilities type

* Move client capabilities to features

* Prepare jupyterlab-lsp 3.10.0 release  (#740)

* Update changelog and bump version to 3.10.0

* Upgrade CodeMirror, lerna and pyright

* Update to JupyterLab 3.1 part 1

* Update devDependencies including JupyterLab packages to 3.2.x

* Mention all bug fixes from null checks PR in the changelog

* Fix jest config

* Lint everything

* Final set of dev dependencies bumps

* Bump versions of modified packages

* Bump range in more places

* Integrity updates

* Fix tooltip types (undefined)

* Final lock file bump

* Lint again

* Update date and lab versions to test on CI

* Implement jump to definitions and basic target selector

* Fix jumping to dependencies in node modules

* Remove unused highlight commands and highlight type icon

* Move files

* Integrate `jupyterlab-lsp` into `jupyterlab`

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Add `jupyter-lsp`

* Update Playwright Snapshots

* Add schema check

* Update Playwright Snapshots

* Remove any type

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update galata tests

* Apply reviewer's suggestions

* Move lsp plugins to required list

* Update staging dependencies

* Revert change on staging/package.json

* Update Playwright Snapshots

* Update Playwright Snapshots.

* Update code structure + add docstring

* Remove `nbformat`

* Apply reviewer's suggestions

* Fix integrity, tests and staging dependencies

* Remove `nbformat` dependency

* Fix failed UI tests

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Update integrity

* Apply reviewer's suggestions

* Package integrity updates

* Lint code

* Tag new APIs as `alpha`

* Rename `initialized` to `ready`

* Add missing header and update setting schema

* Remove `lsp-ws-connection` dependency

* Fix header

* Remove `codemirror` dependency

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* Fix lint

* Apply reviewer's suggestion

Co-authored-by: krassowski <krassowski.michal@gmail.com>
Co-authored-by: Michał Krassowski <5832902+krassowski@users.noreply.github.com>
Co-authored-by: Brendan Powers <brendan0powers@gmail.com>
Co-authored-by: Brendan Powers <bpowers39@bloomberg.net>
Co-authored-by: Serhat YILDIRIM <serhat.yildirim@fity-five.com>
Co-authored-by: JessicaBarh <jessicabarhoma@gmail.com>
Co-authored-by: julioyildo <julioyildo@users.noreply.github.com>
Co-authored-by: Nicholas Bollweg <nick.bollweg@gmail.com>
Co-authored-by: Matthias Bussonnier <bussonniermatthias@gmail.com>
Co-authored-by: Jeremy Tuloup <jeremy.tuloup@gmail.com>
Co-authored-by: passer <whzhoua@gmail.com>
Co-authored-by: zhouwenhua02 <zhouwenhua02@meituan.com>
Co-authored-by: bsyouness <bs.youness@gmail.com>
Co-authored-by: Youness Bennani <ybennani@bloomberg.net>
Co-authored-by: cccs-jc <cccs-jc@cyber.gc.ca>
Co-authored-by: Frédéric Collonval <fcollonval@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

fcollonval committed:
Add license header fix job (#12872)

* Add license header fix job

* Set header content

* Fix unsupported files

* Try pushing manually as down in galata update

* Fix getting the pull request number

* Remove fake git author

* Display commit

* Commit manually

* Add license headers

* Push only if there are changes

* Ignore staging folder

* Revert changes in staging folder

* Ignore integration test files

afshin committed:
Merge pull request #12875 from trungleduc/relax-content-type

Set `Contents.ContentType` to string

dependabot[bot] committed:
Bump toshimaru/auto-author-assign from 1.5.1 to 1.6.0 (#12890)

Bumps [toshimaru/auto-author-assign](https://github.com/toshimaru/auto-author-assign) from 1.5.1 to 1.6.0.
- [Release notes](https://github.com/toshimaru/auto-author-assign/releases)
- [Changelog](https://github.com/toshimaru/auto-author-assign/blob/main/CHANGELOG.md)
- [Commits](https://github.com/toshimaru/auto-author-assign/compare/v1.5.1...v1.6.0)

---
updated-dependencies:
- dependency-name: toshimaru/auto-author-assign
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

fcollonval committed:
Add license header fix to git-blame-ignore (#12900)

jtpio committed:
Split commands in two blocks in the contributing guide (#12898)

* Split commands in two blocks in the contributing guide

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

jtpio committed:
Add `dev_mode/style.js` to the licenser ignore list (#12902)

* Add `dev_mode/style.js` to the licenser ignore list

* Remove license header in dev_mode/style.js

hbcarlos committed:
Fixes renaming files from title while using a custom drive (#12849)

* Fixes renaming files from title while using a custom drive

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

* lint

* Update packages/docregistry/src/context.ts

Fix typo

Co-authored-by: Jeremy Tuloup <jeremy.tuloup@gmail.com>

* Update extension migration guide

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Jeremy Tuloup <jeremy.tuloup@gmail.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] pre-commit autoupdate (#12893)

* [pre-commit.ci] pre-commit autoupdate

updates:
- [github.com/pycqa/flake8: 4.0.1 → 5.0.2](https://github.com/pycqa/flake8/compare/4.0.1...5.0.2)

* Fix flake8 v5 compat

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Steven Silvester <steven.silvester@ieee.org>

afshin committed:
Remove @lumino/coreutils dependency from @jupyterlab/buildutils (#12910)

trungleduc committed:
Log launcher error to console (#12909)

thetorpedodog committed:
Add an option to enable "fast checks" of the jupyter lab build. (#12844)

This adds the `skip_full_build_check` option to the `AppOptions`
configuration. A user can set this option to `true` to avoid performing
expensive work like downloading data from NPM and computing hashes when
the install is in a known-good state or the user cannot actually trigger
a rebuild.

JohanMabille committed:
Moved theme Map and added API to use it from other dependencies (#12861)

* Moved theme Map and added API to use it from other dependencies

* Add test for mode registration

* Add codemirror theme and test

* Improve styling

* Fix code folding for file editors

* Translate internal CM6 strings

* Ensure test grammar is built for tests

* Ignore unsupported .grammar file

* More theme enhancements
Use codemirror theme example to test theme registration

* Fix jest test

* Fix search highlights outside codemirror editors

* Update Playwright Snapshots

* Update Playwright Snapshots

Co-authored-by: Frédéric Collonval <fcollonval@gmail.com>
Co-authored-by: Frédéric Collonval <fcollonval@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

isabela-pf committed:
Apply @krassowski suggestions

Co-authored-by: Michał Krassowski <5832902+krassowski@users.noreply.github.com>

krassowski committed:
Merge branch 'master' into altscipy

krassowski committed:
Add line breaks to fix syntax issue in with comments
